ir/memory.cpp

```diff
@@ -1045,16 +1045,18 @@ pair<expr, expr> Memory::mkUndefInput(unsigned attributes) const {
 }
 
 pair<expr,expr>
-Memory::mkFnRet(const char *name, const vector<StateValue> &ptr_inputs) const {
+Memory::mkFnRet(const char *name,
+                const vector<pair<StateValue, bool>> &ptr_inputs) const {
   expr var
     = expr::mkFreshVar(name, expr::mkUInt(0, bits_for_bid + bits_for_offset));
   Pointer p(*this, var.concat_zeros(bits_for_ptrattrs));
 
   set<expr> local;
   for (auto &in : ptr_inputs) {
-    Pointer inp(*this, in.value);
+    // TODO: callee cannot observe the bid if this is byval.
```

Comment:
Several decisions that are not directly relevant with the example are made conservatively